test/config-spec.js

```diff
@@ -0,0 +1,70 @@
+var assert = require('assert'),
+  uuid = require('node-uuid');
+
+describe('config', function() {
+  describe('retrocompat', function() {
+
+    var retrocompat = require('../src/modules/retrocompat-config');
+
+    it('should set config as empty array if no config present', function() {
+      let conf = retrocompat();
+      assert.deepEqual(conf, {});
+    });
+
+    it('should be able to override default config options based on config object', function() {
+      let conf = {
+        amqpUrl: 'ok',
+        amqpPrefetch: 1,
+        amqpRequeue: true,
+        amqpTimeout: 5
+      };
+
+      retrocompat(conf);
+
+      assert.equal(conf.host, conf.amqpUrl);
+      assert.equal(conf.prefetch, conf.amqpPrefetch);
+      assert.equal(conf.requeue, conf.amqpRequeue);
+      assert.equal(conf.timeout, conf.amqpTimeout);
+    });
+
+    it('should be able to override default config options based on env vars', function() {
+      process.env.LOCAL_QUEUE = ':test';
```

Comment:
Yeah you are right it would be a lot better, and allow us to modify tests at a later date without bothering with test order